llvm/lib/Bitcode/Reader/BitcodeReader.cpp

```diff
@@ -2332,6 +2335,23 @@ Error BitcodeReader::parseAttributeGroupBlock() {
           i--;
 
           B.addConstantRangeAttr(Kind, MaybeCR.get());
+        } else if (Record[i] == 8) {
+          Attribute::AttrKind Kind;
+          if (Error Err = parseAttrKind(Record[++i], &Kind))
+            return Err;
+          if (!Attribute::isConstantRangeListAttrKind(Kind))
+            return error("Not a constant range list attribute");
+
+          SmallVector<ConstantRange, 2> Val;
+          int RangeSize = Record[++i];
+          assert(i + 2 * RangeSize < e);
```

Comment:
This can't be an assert, it should be an if with error return.

Reply:
Ah, changed to an if with error return!